nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ScanAttribute.java

```diff
@@ -151,24 +182,41 @@ public void onScheduled(final ProcessContext context) throws IOException {
 
         this.dictionaryTerms = createDictionary(context);
         this.fileWatcher = new SynchronousFileWatcher(Paths.get(context.getProperty(DICTIONARY_FILE).getValue()), new LastModifiedMonitor(), 1000L);
+
+        this.dictionaryEntryMetadataDemarcator = context.getProperty(DICTIONARY_ENTRY_METADATA_DEMARCATOR).getValue();
     }
 
-    private Set<String> createDictionary(final ProcessContext context) throws IOException {
-        final Set<String> terms = new HashSet<>();
+    private Map<String,String> createDictionary(final ProcessContext context) throws IOException {
+        final Map<String,String> termsMeta = new HashMap<String, String>();
```

Comment:
Please take advantage of diamond operator.

final Map<String,String> termsMeta = new HashMap<>();